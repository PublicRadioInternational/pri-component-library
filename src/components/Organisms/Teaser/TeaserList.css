/* import colors... */
@value colors: "../../00_global/colors.css";
@value grayDark, grayLight, grayLighter, whiteDark from colors;

.section {
  display: flex;
  flex-direction: column;
  max-height: 100vh;
  background-color: whiteDark;
}

.header {
  border-top: 3px solid #ddd;
  border-bottom: 1px solid grayLighter;
  padding: 10px 15px;
  font-size: 1rem;
  line-height: 1.2;
  color: grayDark;
  margin: 0;
  font-family: 'Raleway', sans-serif;
}

.content {
  flex-grow: 1;
  overflow-x: hidden;
  overflow-y: scroll;
}

.content::-webkit-scrollbar {
  /*display: none;*/
  width: 0;
}

.footer {
  padding: 1rem;
  border-top: 1px solid grayLighter;
}

.footer a {
  display: block;
  font-weight: normal;
  text-transform: none;
  box-sizing: border-box;
}

.footer a::after {
  content: ' >';
}
