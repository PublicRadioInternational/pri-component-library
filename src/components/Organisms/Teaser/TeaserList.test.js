/**
 * @file TeaserList.test.js
 * Contains tests for TeaserList.component.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import TeaserList from './TeaserList.component';

describe('<TeaserItem />', () => {
  it('Matches the Teaser List with more link snapshot', () => {
    const component = renderer
      .create(
        <TeaserList title="Test Title" moreTitle="More Stories" moreUrl="more/">
          <span>Test Item One</span>
          <span>test Item Two</span>
        </TeaserList>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Matches the Teaser List without more link snapshot', () => {
    const component = renderer
      .create(
        <TeaserList title="Test Title">
          <span>Test Item One</span>
          <span>test Item Two</span>
        </TeaserList>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
