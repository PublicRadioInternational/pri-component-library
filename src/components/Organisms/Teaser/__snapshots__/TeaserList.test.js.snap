// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeaserItem /> Matches the Teaser List with more link snapshot 1`] = `
<section
  className="className && section"
>
  <header
    className="header"
  >
    Test Title
  </header>
  <div
    className="content"
  >
    <span>
      Test Item One
    </span>
    <span>
      test Item Two
    </span>
  </div>
  <footer
    className="footer"
  >
    <a
      className="btnBlue"
      href="more/"
      onClick={[Function]}
    >
      More Stories
    </a>
  </footer>
</section>
`;

exports[`<TeaserItem /> Matches the Teaser List without more link snapshot 1`] = `
<section
  className="className && section"
>
  <header
    className="header"
  >
    Test Title
  </header>
  <div
    className="content"
  >
    <span>
      Test Item One
    </span>
    <span>
      test Item Two
    </span>
  </div>
</section>
`;
